'invalid memory address or nil pointer dereference' error in Datadog Logs exporter

### Component(s)

exporter/datadog

### What happened?

## Description

`invalid memory address or nil pointer dereference` runtime error happened for no apparent reason in the Datadog Logs exporter as per the trace info.

## Steps to Reproduce

Unknown (we have the collector running in the VM and when I logged into the machine today to check something else, it was stopped with this error in the console).

## Expected Result

Collector should run continuously without ever producing errors like this.

## Actual Result

Collector completely halted execution.

### Collector version

0.62.1

### Environment information

## Environment
Azure Windows VM

### OpenTelemetry Collector configuration

```yaml
receivers:
  otlp:
    protocols:
      grpc:

  hostmetrics:
    collection_interval: 10s
    scrapers:
      paging:
        metrics:
          system.paging.utilization:
            enabled: true
      cpu:
        metrics:
          system.cpu.utilization:
            enabled: true
      disk:
      filesystem:
        metrics:
          system.filesystem.utilization:
            enabled: true
      load:
      memory:
      network:
      processes:

processors:
  batch:
    timeout: 10s

  resource:
    attributes:
    - key: deployment.environment
      value: "beta"
      action: upsert

exporters:
  datadog:
    api:
      key: ${DATADOG_APIKEY}

service:
  pipelines:
    traces:
      receivers: [otlp]
      processors: [resource, batch]
      exporters: [datadog]
    metrics:
      receivers: [hostmetrics, otlp]
      processors: [resource, batch]
      exporters: [datadog]
    logs:
      receivers: [otlp]
      processors: [resource, batch]
      exporters: [datadog]
```


### Log output

```shell
2022-10-27T07:53:56.703Z        info    metadata/metadata.go:216        Sent host metadata      {"kind": "exporter", "data_type": "metrics", "name": "datadog"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x40 pc=0x29487f6]

goroutine 165 [running]:
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/logs.(*Sender).SubmitLogs(0xc0008278e0, {0x78ea758, 0xc00115bd00}, {0xc001f93b00, 0x2, 0x2})
        github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter@v0.62.0/internal/logs/sender.go:67 +0x236
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter.(*logsExporter).consumeLogs(0xc000bf1b80, {0x6cbc76e?, 0x10?}, {0xc001f93a80?})
        github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter@v0.62.0/logs_exporter.go:101 +0x16c
go.opentelemetry.io/collector/exporter/exporterhelper.(*logsRequest).Export(0x2ed8f1e?, {0x78ea800?, 0xc002023b00?})
        go.opentelemetry.io/collector@v0.62.1/exporter/exporterhelper/logs.go:65 +0x34
go.opentelemetry.io/collector/exporter/exporterhelper.(*timeoutSender).send(0xc000962490, {0x7907d80, 0xc000b90c30})
        go.opentelemetry.io/collector@v0.62.1/exporter/exporterhelper/common.go:203 +0x96
go.opentelemetry.io/collector/exporter/exporterhelper.(*retrySender).send(0xc00069d440, {0x7907d80, 0xc000b90c30})
        go.opentelemetry.io/collector@v0.62.1/exporter/exporterhelper/queued_retry.go:388 +0x58d
go.opentelemetry.io/collector/exporter/exporterhelper.(*logsExporterWithObservability).send(0xc000c62198, {0x7907d80, 0xc000b90c30})
        go.opentelemetry.io/collector@v0.62.1/exporter/exporterhelper/logs.go:132 +0x88
go.opentelemetry.io/collector/exporter/exporterhelper.(*queuedRetrySender).start.func1({0x7907d80, 0xc000b90c30})
        go.opentelemetry.io/collector@v0.62.1/exporter/exporterhelper/queued_retry.go:206 +0x39
go.opentelemetry.io/collector/exporter/exporterhelper/internal.(*boundedMemoryQueue).StartConsumers.func1()
        go.opentelemetry.io/collector@v0.62.1/exporter/exporterhelper/internal/bounded_memory_queue.go:61 +0xb6
created by go.opentelemetry.io/collector/exporter/exporterhelper/internal.(*boundedMemoryQueue).StartConsumers
        go.opentelemetry.io/collector@v0.62.1/exporter/exporterhelper/internal/bounded_memory_queue.go:56 +0x45
```


### Additional context

_No response_

--- a/.chloggen/mx-psi_logs-exporter-crash.yaml
+++ b/.chloggen/mx-psi_logs-exporter-crash.yaml
@@ -1,0 +1,11 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: datadogexporter
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Fixes crash when logging error on logs exporter
+
+# One or more tracking issues related to the change
+issues: [16077]

--- a/exporter/datadogexporter/internal/logs/sender.go
+++ b/exporter/datadogexporter/internal/logs/sender.go
@@ -20,6 +20,7 @@ import (
 
 	"github.com/DataDog/datadog-api-client-go/v2/api/datadog"
 	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV2"
+	"go.opentelemetry.io/collector/consumer/consumererror"
 	"go.opentelemetry.io/collector/exporter/exporterhelper"
 	"go.uber.org/zap"
 
@@ -73,10 +74,16 @@ func (s *Sender) SubmitLogs(ctx context.Context, payload []datadogV2.HTTPLogItem
 	}
 	_, r, err := s.api.SubmitLog(ctx, payload, s.opts)
 	if err != nil {
-		b := make([]byte, 1024) // 1KB message max
-		n, _ := r.Body.Read(b)  // ignore any error
-		s.logger.Error("Failed to send logs", zap.Error(err), zap.String("msg", string(b[:n])), zap.String("status_code", r.Status))
-		return err
+		if r != nil {
+			b := make([]byte, 1024) // 1KB message max
+			n, _ := r.Body.Read(b)  // ignore any error
+			s.logger.Error("Failed to send logs", zap.Error(err), zap.String("msg", string(b[:n])), zap.String("status_code", r.Status))
+			return err
+		}
+
+		// If response is nil assume permanent error.
+		// The error will be logged by the exporter helper.
+		return consumererror.NewPermanent(err)
 	}
 	return nil
 }

